reddit_decider/__init__.py

```diff
@@ -1026,30 +1004,32 @@ def _prune_extracted_dict(extracted_dict: dict) -> dict:
         return parsed_extracted_fields
 
     def _minimal_decider(
-        self, name: str, span: Span, parsed_extracted_fields: Optional[Dict] = None
+        self,
+        rs_decider: Optional[RustDecider],
+        name: str,
+        span: Span,
+        parsed_extracted_fields: Optional[Dict] = None,
     ) -> Decider:
         return Decider(
             decider_context=DeciderContext(extracted_fields=parsed_extracted_fields),
-            config_watcher=self._filewatcher,
+            rs_decider=rs_decider,
             server_span=span,
             context_name=name,
             event_logger=self._event_logger,
         )
 
     def make_object_for_context(self, name: str, span: Span) -> Decider:
-        decider = None
+        rs_decider = None
         try:
-            decider = self._filewatcher.get_data()
+            rs_decider = self._filewatcher.get_data()
         except WatchedFileNotAvailableError as exc:
-            logger.error("Experiment config file unavailable: %s", str(exc))
-        except TypeError as exc:
-            logger.error("Could not load experiment config: %s", str(exc))
-
-        validate_decider(decider)
+            logger.error(f"Experiment config file unavailable: {exc}")
+        except DeciderInitException as exc:
+            logger.error(f"Could not load experiment config: {exc}")
```

Comment:
should I add a general:
`except Exception as exc:`
block, so if `RustDecider` init fails, we don't bubble up an exception and just initialize a SDK `Decider` instance without breaking the request? (errors will log e.g. [here](https://github.com/reddit/experiments.py/pull/79/files#diff-00f6e0196121fdac7fda7102cc51166ba1e97b773541f1b7788f34dd6637c3f9R342) on calls to `Decider` api)

Reply:
What calls this method? It's generally better for the calling code to handle an exception rather than to silently swallow the exception in the client.

Reply:
`make_object_for_context()` gets called internally by baseplate, so we'll want to rescue this so as to not take down the request fully. 
the filewatcher covers all exceptions with `WatchedFileNotAvailableError` [here](https://github.com/reddit/baseplate.py/blob/bc2d21fdc259a785ef5bf14d1334fbd4ea1623ce/baseplate/lib/file_watcher.py#L199) though, so we should be good with just covering this exception.